src/Plugin/rest/resource/YouTubeRestResource.php

```diff
@@ -0,0 +1,129 @@
+<?php
+
+namespace Drupal\youtube_proxy\Plugin\rest\resource;
+
+use Drupal\rest\Plugin\ResourceBase;
+use Drupal\rest\ResourceResponse;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+use Symfony\Component\HttpKernel\Exception\AccessDeniedHttpException;
+use Symfony\Component\HttpKernel\Exception\UnprocessableEntityHttpException;
+use Symfony\Component\HttpKernel\Exception\NotFoundHttpException;
+
+/**
+ * Provides a resource to get view modes by entity and bundle.
+ *
+ * @RestResource(
+ *   id = "youtube_rest_resource",
+ *   label = @Translation("YouTube rest resource"),
+ *   uri_paths = {
+ *     "canonical" = "/api/yt_proxy/{video_id}"
+ *   }
+ * )
+ */
+class YouTubeRestResource extends ResourceBase {
+
+  /**
+   * A current user instance.
+   *
+   * @var \Drupal\Core\Session\AccountProxyInterface
+   */
+    protected $currentUser;
+    private $_cacheId;
```

Comment:
Wordt `$_cacheId` nog gebruikt? zie het niet in die bestand...
Benaming is ook niet echt netjes met die underscore aan het begin.

Reply:
$_cacheId wordt wat lager in het bestand meerdere malen gebruikt regels: 80, 112, 115. 
PHPCS geeft aan dat je voor private variables underscores moet gebruiken: "Private member variable "cacheId" must be prefixed with an underscore".